src/pages/users/AddUsers.vue

```diff
@@ -252,190 +267,89 @@ const onFileUpload = async (event) => {
   }
 
   // Store the parsed data
-  rawUserFile.value = parsedData;
+  rawUserFile.value = filteredData;
 
-  // REGISTRATION
-  // Required: userType
-  // Conditional (child): Month, Year
-  // Conditional (Either): Cohort OR Site + School
-
-  // Get all column names from the first row, case-insensitive check for userType
   const firstRow = toRaw(rawUserFile.value[0]);
-  const allColumns = Object.keys(firstRow).map((col) => col.toLowerCase());
+  const headers = Object.keys(firstRow);
 
-  // Check if userType column exists (case-insensitive)
-  const hasUserType = allColumns.includes('usertype');
-  if (!hasUserType) {
-    toast.add({
-      severity: 'error',
-      summary: 'Error: Missing Column',
-      detail: 'Missing required column(s): userType',
-      life: TOAST_DEFAULT_LIFE_DURATION,
-    });
-    errorMissingColumns.value = true;
-    return;
-  }
-
-  // Check conditional columns are present
-  const hasChild = rawUserFile.value.some((user) => {
-    const userTypeValue = Object.keys(user).find((key) => key.toLowerCase() === 'usertype');
-    return userTypeValue && user[userTypeValue].toLowerCase() === 'child';
+  const requiredHeaders = ['usertype'];
+  const hasChild = filteredData.some((user) => {
+    const userTypeField = Object.keys(user).find((key) => key.toLowerCase() === 'usertype');
+    return userTypeField && user[userTypeField]?.toLowerCase() === 'child';
   });
 
-  // If we have child users, they MUST have month and year
   if (hasChild) {
-    const hasMonth = allColumns.includes('month');
-    const hasYear = allColumns.includes('year');
-    if (!hasMonth || !hasYear) {
-      toast.add({
-        severity: 'error',
-        summary: 'Error: Missing Column',
-        detail: 'Missing required column(s): Month or Year',
-        life: TOAST_DEFAULT_LIFE_DURATION,
-      });
-      errorMissingColumns.value = true;
-      return;
-    }
+    requiredHeaders.push('month', 'year');
   }
 
-  // Conditional (Either): Cohort OR Site + School
-  const hasCohort = allColumns.includes('cohort');
-  const hasSchool = allColumns.includes('school');
+  const hasCohort = headers.some((col) => col.toLowerCase() === 'cohort');
+  const hasSchool = headers.some((col) => col.toLowerCase() === 'school');
```

Comment:
Since we do these lowercase checks everywhere, we should get the headers once, parse them all to lowercase, and then just work off of that so we don't keep doing this every time we're working with the headers.